### ISSUE TITLE
orb-babel-async-language-blacklist not defined

### ISSUE THREAD

**Issue by benneti** [hlissner/doom-emacs#4726]
**What did you expect to happen?**
Org-babel should execute a cell on C-c C-c

**What actually happened?**
void variable in the `+org-babel-disable-async-if-needed-a`

**Additional details:**
- https://cloud.tissot.de/gitea/benneti/doom
- org-ctrl-c-ctrl-c

<details><pre>
Debugger entered--Lisp error: (void-variable org-babel-async-language-blacklist)
  (member (car info) org-babel-async-language-blacklist)
  (or (assq :sync params) (not (assq :async params)) (member (car info) org-babel-async-language-blacklist))
  (cond ((or (assq :sync params) (not (assq :async params)) (member (car info) org-babel-async-language-blacklist)) (funcall fn arg info params)) ((not (member (cdr (assq :session params)) '("none" nil))) (message "Org babel :: :session is incompatible with :async....") nil) ((funcall orig-fn fn arg info params)))
  (let* ((info (or info (org-babel-get-src-block-info))) (params (org-babel-merge-params (nth 2 info) params))) (cond ((or (assq :sync params) (not (assq :async params)) (member (car info) org-babel-async-language-blacklist)) (funcall fn arg info params)) ((not (member (cdr (assq :session params)) '("none" nil))) (message "Org babel :: :session is incompatible with :async....") nil) ((funcall orig-fn fn arg info params))))
  (if (null fn) (funcall orig-fn fn arg info params) (let* ((info (or info (org-babel-get-src-block-info))) (params (org-babel-merge-params (nth 2 info) params))) (cond ((or (assq :sync params) (not (assq :async params)) (member (car info) org-babel-async-language-blacklist)) (funcall fn arg info params)) ((not (member (cdr (assq :session params)) '("none" nil))) (message "Org babel :: :session is incompatible with :async....") nil) ((funcall orig-fn fn arg info params)))))
  +org-babel-disable-async-if-needed-a(#<subr ob-async-org-babel-execute-src-block> #f(advice-wrapper :around #<subr org-babel-execute-src-block> +direnv-persist-environment-a) nil ("jupyter-julia" "fname = datadir(\"dissipative_timetraces_L=100\", \"T..." ((:colname-names) (:rowname-names) (:result-params "raw" "replace") (:result-type . value) (:results . "raw replace") (:exports . "both") (:session . "ttlong") (:eval . "no-export") (:async . "yes") (:kernel . "julia-1.5") (:tangle . "no") (:hlines . "no") (:noweb . "no") (:cache . "no")) "" nil 517 "(ref:%s)"))
  apply(+org-babel-disable-async-if-needed-a #<subr ob-async-org-babel-execute-src-block> (#f(advice-wrapper :around #<subr org-babel-execute-src-block> +direnv-persist-environment-a) nil ("jupyter-julia" "fname = datadir(\"dissipative_timetraces_L=100\", \"T..." ((:colname-names) (:rowname-names) (:result-params "raw" "replace") (:result-type . value) (:results . "raw replace") (:exports . "both") (:session . "ttlong") (:eval . "no-export") (:async . "yes") (:kernel . "julia-1.5") (:tangle . "no") (:hlines . "no") (:noweb . "no") (:cache . "no")) "" nil 517 "(ref:%s)")))
  ob-async-org-babel-execute-src-block(#f(advice-wrapper :around #<subr org-babel-execute-src-block> +direnv-persist-environment-a) nil ("jupyter-julia" "fname = datadir(\"dissipative_timetraces_L=100\", \"T..." ((:colname-names) (:rowname-names) (:result-params "raw" "replace") (:result-type . value) (:results . "raw replace") (:exports . "both") (:session . "ttlong") (:eval . "no-export") (:async . "yes") (:kernel . "julia-1.5") (:tangle . "no") (:hlines . "no") (:noweb . "no") (:cache . "no")) "" nil 517 "(ref:%s)"))
  apply(ob-async-org-babel-execute-src-block #f(advice-wrapper :around #<subr org-babel-execute-src-block> +direnv-persist-environment-a) (nil ("jupyter-julia" "fname = datadir(\"dissipative_timetraces_L=100\", \"T..." ((:colname-names) (:rowname-names) (:result-params "raw" "replace") (:result-type . value) (:results . "raw replace") (:exports . "both") (:session . "ttlong") (:eval . "no-export") (:async . "yes") (:kernel . "julia-1.5") (:tangle . "no") (:hlines . "no") (:noweb . "no") (:cache . "no")) "" nil 517 "(ref:%s)")))
  org-babel-execute-src-block(nil ("jupyter-julia" "fname = datadir(\"dissipative_timetraces_L=100\", \"T..." ((:colname-names) (:rowname-names) (:result-params "raw" "replace") (:result-type . value) (:results . "raw replace") (:exports . "both") (:session . "ttlong") (:eval . "no-export") (:async . "yes") (:kernel . "julia-1.5") (:tangle . "no") (:hlines . "no") (:noweb . "no") (:cache . "no")) "" nil 517 "(ref:%s)"))
  org-ctrl-c-ctrl-c(nil)
  #<subr funcall-interactively>(org-ctrl-c-ctrl-c nil)
  apply(#<subr funcall-interactively> (org-ctrl-c-ctrl-c nil))
  funcall-interactively(org-ctrl-c-ctrl-c nil)
  #<subr call-interactively>(org-ctrl-c-ctrl-c nil nil)
  apply(#<subr call-interactively> (org-ctrl-c-ctrl-c nil nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> org-ctrl-c-ctrl-c nil nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (org-ctrl-c-ctrl-c nil nil))
  call-interactively(org-ctrl-c-ctrl-c nil nil)
  command-execute(org-ctrl-c-ctrl-c)

</pre></details>

**Steps to reproduce:**
1. enable org (for me +jupyter) module
2. create an org file with a babel cell (for me julia-jupyter)
3. execute one cell (should be fine)
4. create and execute a new cell, run into trouble with the async advice
5. fix it by defining the variable as  described in **FIX**

**System information:**
<details><pre>
```
SYSTEM  info       (NixOS 21.05.20210213.ff96a0f (Okapi) . Linux 5.10.15 x86_64)
        shell      /run/current-system/sw/bin/zsh
        path       (/run/wrappers/bin ~/.nix-profile/bin /etc/profiles/per-user/$USER/bin /nix/var/nix/profiles/default/bin /run/current-system/sw/bin /nix/store/bpsbkj8acws4jlxxcs9q5gx64jxshf5x-fzf-0.25.0/bin ~/.config/emacs/bin /nix/store/7hpw2jvqb3y15vhrvzhbzs19ycf6ga5g-emacs-pgtkgcc-20210302.0/libexec/emacs/28.0.50/x86_64-pc-linux-gnu/)
EMACS   dir        ~/.config/emacs/
        version    (28.0.50 45129d48fbbc2432a6010efe032d4fda34258e40 Mar 03, 2021)
        buildopts  --prefix=/nix/store/7hpw2jvqb3y15vhrvzhbzs19ycf6ga5g-emacs-pgtkgcc-20210302.0 --disable-build-details --with-modules --with-x-toolkit=gtk3 --with-cairo --with-native-compilation --with-pgtk
        features   CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GSETTINGS HARFBUZZ JPEG JSON LIBOTF LIBSELINUX LIBSYSTEMD LIBXML2 MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PGTK PNG RSVG SOUND THREADS TIFF TOOLKIT_SCROLL_BARS XIM GTK3 ZLIB
        traits     (gui server-running envvar-file)
DOOM    dir        ~/.config/doom/
        version    (2.0.9 HEAD -> develop d5019f0b5 2021-03-02 23:48:54 -0500)
        font       #<font-spec nil nil monospace nil nil nil nil nil 20 nil nil nil nil>
        theme      doom-dracula
        modules    (:completion company (ivy +icons) :ui doom doom-dashboard doom-quit hl-todo modeline ophints (popup +defaults) ligatures unicode vc-gutter vi-tilde-fringe (window-select +numbers +switch-window) :editor (evil +everywhere) file-templates fold rotate-text snippets word-wrap :emacs (dired +ranger +icons) electric (ibuffer +icons) undo vc :term vterm :checkers syntax (spell +everywhere) grammar :tools direnv editorconfig (eval +overlay) lookup (lsp +eglot) magit (pass +auth) pdf biblio :lang (cc +lsp) data emacs-lisp (julia +lsp) (latex +latexmk +cdlatex +fold) markdown nix (org +jupyter +dragndrop +pandoc +pretty +ref +roam) (python +lsp) sh :email mu4e :config (default +bindings +smartparens))
        packages   ((company-statistics) (company-quickhelp) (academic-phrases) (org-ref) (latex-preview-pane :disable) (org-ql) (svg-tag-mode :recipe (:host github :repo rougier/svg-tag-mode :branch main :files (svg-tag-mode.el))) (langtool-ignore-fonts :recipe (:host github :repo cjl8zf/langtool-ignore-fonts :files (langtool-ignore-fonts.el))) (org-appear :recipe (:host github :repo awth13/org-appear)) (org-caldav) (mixed-pitch))
        unpin      (n/a)
        elpa       (n/a)
```

</pre></details>

**FIX**
(setq org-babel-async-language-blacklist '("jupyter-julia" "jupyter-python"))

I think the intended behaviour is to let jupyter and the like handle the async option themselfe.
A more permanent fix would be to either change the advice `+org-babel-disable-async-if-needed-a` in the org module or add a default value there.

If you would tell me what you prefer I would be happy to create a PR.

**Comment by hlissner** [comment]
My bad. The variable is supposed to be `ob-async-no-async-languages-alist`. This is fixed in 2ab8800. Thanks for bringing it to my attention!

**Comment by benneti** [comment]
thanks for the quick fix!
